### PR TITLE
Add travis script

### PR DESCRIPTION
**Pull request by jasonlaska**
See if this is working.

### CODE REVIEW

**Comment by jasonlaska** [comment]
Woo.  Now need to figure out the numerical issues with the tests.

**Comment by mnarayan** [comment]
Build fails due to numpy installation failure
```
`The command "pip install -r requirements.txt" failed
```

Maybe we'll just edit to have conda installations to start with. 
```
{
  "language": "python",
  "python": "2.7",
  "env": "DISTRIB=\"ubuntu\" PYTHON_VERSION=\"2.7\" CYTHON_VERSION=\"0.23.4\" COVERAGE=true",
  "install": [
    "pip install -r requirements.txt"
  ],
  "script": "pytest inverse_covariance",
  "group": "stable",
  "dist": "precise",
  "os": "linux"
}
```

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    packages:
+        - libatlas3gf-base
+        - libatlas-dev
+        - python-scipy
+
+# command to install dependencies
+install:
+    - pip install -U pip wheel
+    - pip install -r requirements.txt
+    - python setup.py develop
```
Comment (jasonlaska) [inline .travis.yml]: You probably either want to just run `setup.py` here or change to the inverse_covariance/pyquic directory and run `make`

Comment (jasonlaska) [inline .travis.yml]: if just setup.py works, I think that's preferred

File: test_script.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+set -e
+
+python --version
+python -c "import numpy; print('numpy %s' % numpy.__version__)"
+python -c "import scipy; print('scipy %s' % scipy.__version__)"
+python -c "import sklearn; print('sklearn %s' % sklearn.__version__)"
+
+
+python -m pytest --showlocals --pyargs
```
Comment (jasonlaska) [inline test_script.sh]: is there a reason this can't be run directly in the YAML file?

Comment (jasonlaska) [inline test_script.sh]: also we just want to run the tests in `inverse_covariance` so should be 

```
python -m pytest inverse_covariance --showlocals --pyargs
```

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    apt: true
+    directories:
+        - $HOME/.cache/pip
+
+addons:
+    apt:
+    packages:
+        - libatlas3gf-base
+        - libatlas-dev
+        - python-scipy
```
Comment (jasonlaska) [inline .travis.yml]: try adding cython

**Comment by mnarayan** [comment]
@jasonlaska Take a look at this https://github.com/cjrh/easycython

**Comment by jasonlaska** [comment]
That's cool (easy cython).  Maybe make a ticket about it.  Since our setup.py is already good to go, don't see a need in changing right now.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        - cython
+
+# command to install dependencies
+install:
+    - pip install -U pip wheel
+    - pip install -r requirements.txt
+    - python setup.py build_ext --inplace
+    - python setup.py develop
```
Comment (jasonlaska) [inline .travis.yml]: i think this second like `setup.py develop` is unneeded

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+        - libatlas3gf-base
+        - libatlas-dev
+        - python-scipy
+        - cython
+
+# command to install dependencies
+install:
+    - pip install -U pip wheel
+    - pip install -r requirements.txt
+    - python setup.py build_ext --inplace
```
Comment (jasonlaska) [inline .travis.yml]: I don't know if you need `build_ext` or `--inplace`.  I don't need those params when i run locally.

**Comment by mnarayan** [comment]
Also it looks like other scikit projects run into this pyx build issue 
```pip install --install-option="--no-cython-compile" cython```. Recommended in https://github.com/symengine/symengine/issues/308

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        - libatlas3gf-base
+        - libatlas-dev
+        - python-scipy
+        - cython
+
+# command to install dependencies
+install:
+    - pip install -U pip wheel
+    - pip install -r requirements.txt
+    - pip install --upgrade --install-option="--no-cython-compile" cython
```
Comment (jasonlaska) [inline .travis.yml]: cython is in the requirements so maybe try this line first

File: setup.py [hunk 7]
```diff
@@ -69,5 +71,5 @@
     url='https://github.com/skggm/skggm',
     author_email='jlaska@gmail.com',
     ext_package='inverse_covariance',
-    ext_modules=[ext_module],
-)
\ No newline at end of file
+    ext_modules= cythonize(ext_module),
```
Comment (mnarayan) [inline setup.py]: Using `cythonize` seems to have fixed the travis build issues on this point.

**Comment by mnarayan** [comment]
@jasonlaska so tests are failing due to tiny numerical discrepencies.

**Comment by jasonlaska** [comment]
Some notes before merge:
- all of the tests pass, although some appear to be non-deterministic, which is annoying, so that's a TODO
- there are warnings in `common_test` that I dont think we can avoid
- there are warnings due to nested `Parallel` in MonteCarloProfile that we should work out

**Pull request merged**